code/modules/atmospherics/machinery/airalarm.dm

```diff
@@ -884,6 +887,87 @@
 		new /obj/item/stack/cable_coil(loc, 3)
 	qdel(src)
 
+/obj/item/circuit_component/air_alarm
+	display_name = "Air Alarm"
+	desc = "Controls levels of gases and their temperature as well as all vents and scrubbers in the room."
+
+	var/datum/port/input/min_2
+	var/datum/port/input/min_1
+	var/datum/port/input/max_1
+	var/datum/port/input/max_2
+
+	var/datum/port/input/request_data
+
+	var/datum/port/output/pressure
+	var/datum/port/output/temperature
+	var/datum/port/output/gas_amount
+
+	var/obj/machinery/airalarm/connected_alarm
+	var/list/options_map
+
+/obj/item/circuit_component/air_alarm/Initialize()
+	. = ..()
+	min_2 = add_input_port("Min 2", PORT_TYPE_NUMBER)
+	min_1 = add_input_port("Min 1", PORT_TYPE_NUMBER)
+	max_1 = add_input_port("Max 1", PORT_TYPE_NUMBER)
+	max_2 = add_input_port("Max 2", PORT_TYPE_NUMBER)
+	request_data = add_input_port("Request Atmosphere Data", PORT_TYPE_SIGNAL)
+
+	pressure = add_output_port("Pressure", PORT_TYPE_NUMBER)
+	temperature = add_output_port("Temperature", PORT_TYPE_NUMBER)
+	gas_amount = add_output_port("Chosen Gas Amount", PORT_TYPE_NUMBER)
+
+/obj/item/circuit_component/air_alarm/populate_options()
+	var/static/list/component_options
+	var/static/list/options_to_key
+
+	if(!component_options)
+		component_options = list(
+			"Pressure",
+			"Temperature"
+		)
+		options_to_key = list(
+			"Pressure" = "pressure",
+			"Temperature" = "temperature"
+		)
+
+		for(var/datum/gas/gas_type as anything in subtypesof(/datum/gas))
+			component_options.Add(initial(gas_type.name))
+			options_to_key[gas_type.name] = gas_type
```

Comment:
You should just use `GLOB.meta_gas_info` and `META_GAS_NAME` here. They exist for exactly this use, and I don't like anything outside of core atmos code/gaslist memes that treats gastypes as things.

Reply:
ah yeah thats it